compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -647,25 +657,36 @@ impl std::ops::DerefMut for TyAndNaiveLayout<'_> {
 pub struct NaiveLayout {
     pub min_size: Size,
     pub min_align: Align,
+    // If `true`, `min_size` and `min_align` are guaranteed to be exact.
+    pub is_exact: bool,
 }
 
 impl NaiveLayout {
-    pub const EMPTY: Self = Self { min_size: Size::ZERO, min_align: Align::ONE };
+    pub const UNKNOWN: Self = Self { min_size: Size::ZERO, min_align: Align::ONE, is_exact: false };
+    pub const EMPTY: Self = Self { min_size: Size::ZERO, min_align: Align::ONE, is_exact: true };
+
+    pub fn is_compatible_with(&self, layout: Layout<'_>) -> bool {
```

Comment:
This now needs documentation, as it isn't as "obvious" anymore as `is_underestimate_of`.